integration-tests/maven/build.gradle.kts

```diff
@@ -16,10 +14,13 @@ dependencies {
 tasks.integrationTest {
     dependsOnMavenLocalPublication()
 
-    val setupMavenTask = project(":runners:maven-plugin").tasks.withType<SetupMaven>().single()
-    dependsOn(setupMavenTask)
+    dependsOn(tasks.installMavenBinary)
+    val mvn = setupMavenProperties.mvn
+    inputs.file(mvn)
 
     val dokka_version: String by project
     environment("DOKKA_VERSION", dokka_version)
-    environment("MVN_BINARY_PATH", setupMavenTask.mvn.absolutePath)
+    doFirst {
```

Comment:
Ah yes, I forgot to put a comment here. I wanted to find the Gradle issue and put a link in there...


I updated the `mvn` exec file location property to be a [Gradle file property](https://docs.gradle.org/current/userguide/lazy_configuration.html#working_with_files_in_lazy_properties) (which implements the `Provider<T>` API). 

`Exec.environment()` hasn't been updated to be compatible with the [Gradle Provider API](https://docs.gradle.org/current/userguide/lazy_configuration.html#lazy_properties). Providers shouldn't be computed (using `.get()`) until the task is running.

As a workaround, I put the `.get()` in a `doFirst {}`. 

In this case, I am perhaps being a bit too strict. The `mvn` property is static and won't change, so it doesn't have to be a Gradle file property (it could just be a `java.io.File`), or it's fine if in this instance it's computed early by calling `.get()` in the task-config block.

```kotlin
tasks.integrationTest {
    // call .get() in the configuration-phase
    environment("MVN_BINARY_PATH", mvn.get().asFile.invariantSeparatorsPath)
}
```